### ISSUE TITLE
Results do not match Lumerical

### ISSUE THREAD

**Issue by joamatab** [flexcompute/tidy3d#411]
Hi!

How can we make the FDTD simulations match lumerical FDTD?

Im running tidy3d-beta==1.4.0

See Sparameters for a 1x2 mmi in tidy3d

![image](https://user-images.githubusercontent.com/4514346/173466096-a77b8a2c-2356-4327-ad47-655d4a81e429.png)

here is what i think is happening. At one wavelength the mode solver is finding the wrong mode

![image](https://user-images.githubusercontent.com/4514346/173472464-b6ffb8ad-1793-40c3-9753-3f6085418bc9.png)

**Comment by momchil-flex** [comment]
So it seems that it's just that the mode plane covers a part of the other waveguide. By default, we have PEC boundaries at the edges of the mode solver. So this essentially makes a spurious waveguide which I guess turns out to have a higher effective index than the central waveguide at that frequency. Have you actually done the exact same simulation with Lumerical? What boundary conditions do you use for the edges of the mode plane?

More practically, the modes you're looking at seem very strongly confined. It seems that you can decrease the mode plane span along y (and z, just cause why not) without a loss in the accuracy, and this should fix the problem. Let me know!

**Comment by joamatab** [comment]
even when I make the y span = 0.2, which computes the correct modes
![image](https://user-images.githubusercontent.com/4514346/173591480-cb0ccdec-ca8d-4482-aac7-3d743a71f0a7.png)

the Sparameters have some issues

![image](https://user-images.githubusercontent.com/4514346/173591721-2159b78a-c490-4bd5-9dfe-bb36d0a49040.png)

**Comment by momchil-flex** [comment]
The mode source in ports 2 and 3 is injecting in the wrong direction. It's a bit hard to see the arrow here but you can see it once you know it, or you can check s.sources and see that direction is "+".

![image](https://user-images.githubusercontent.com/92756559/173675035-facbeb63-66a2-4728-93a7-4217997c625b.png)

What is happening is that there's a small numerical-error power injected backwards, but since you normalize everything to the total power backwards (which is very small), results look like they *kinda* make sense, but there's a big signal to noise ratio.

Bottom line: this is a simulation setup issue in gdsfactory, so I'm closing this.

**Comment by joamatab** [comment]
Hi!

i fixed the port orientation but the strange spectrum persists

![image](https://user-images.githubusercontent.com/4514346/173722376-7b485428-ad78-4f3b-a6f1-e45758f6e627.png)

**Comment by joamatab** [comment]
https://github.com/gdsfactory/gdsfactory/pull/453

**Comment by momchil-flex** [comment]
I ran this with the latest version of gdsfactory and `port_margin=0.6` and the results look good:
![image](https://user-images.githubusercontent.com/92756559/173893527-088c8fbb-048c-4005-a818-06d814f45744.png)

Same results with `port_margin = 0.9` actually.

How do you clean cached results in `gdsfactory`? If I try to rerun this now (e.g. changing the port margin), it just loads the data from the previous run. To force it to rerun I changed the wavelength range. It seems that in the last image you sent, the cached results were loaded. Maybe that's why it still looks bad, cause it didn't rerun with the updated source orientation?
